internal/repository/sync.go

```diff
@@ -58,5 +60,18 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 	}
 
 	repository := &Repository{r: r}
+	if len(expectedRootKeys) > 0 {
+		state, err := policy.LoadCurrentState(ctx, r)
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+		rootMetadata, err := state.GetRootMetadata()
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+		if !reflect.DeepEqual(rootMetadata.Roles[policy.RootRoleName].KeyIDs, expectedRootKeys) {
```

Comment:
I still don't like using only the key IDs here. Can we instead do a more thorough examination of the key objects loaded vs what's in the keys dir in the policy state?